src/io/flutter/sdk/FlutterCommand.java

```diff
@@ -245,23 +260,57 @@ public OSProcessHandler startProcessOrShowError(@Nullable Project project) {
   public GeneralCommandLine createGeneralCommandLine(@Nullable Project project) {
     final GeneralCommandLine line = new GeneralCommandLine();
     line.setCharset(CharsetToolkit.UTF8_CHARSET);
-
-    line.withEnvironment(FlutterSdkUtil.FLUTTER_HOST_ENV, FlutterSdkUtil.getFlutterHostEnvValue());
-
-    final String androidHome = IntelliJAndroidSdk.chooseAndroidHome(project, false);
-    if (androidHome != null) {
-      line.withEnvironment("ANDROID_HOME", androidHome);
+    if (isFlutterWeb) {
+      if (workDir != null) {
+        line.setWorkDirectory(workDir.getPath());
+      }
+      if (localWebDevExe != null || !localWebDevExe.isEmpty()) {
+        line.setExePath(localWebDevExe);
+        line.addParameters("daemon");
+        if (args.contains("--debug")) {
+          line.addParameters("--debug");
+        }
+        if (args.contains("--hot-reload")) {
+          line.addParameters("--hot-reload");
+        }
+      }
+      else {
+        line.setExePath(FileUtil.toSystemDependentName(sdk.getHomePath() + "/bin/" + FlutterSdkUtil.flutterScriptName()));
+        line.addParameters(args);
+      }
     }
+    else {
+      line.withEnvironment(FlutterSdkUtil.FLUTTER_HOST_ENV, FlutterSdkUtil.getFlutterHostEnvValue());
+      final String androidHome = IntelliJAndroidSdk.chooseAndroidHome(project, false);
+      if (androidHome != null) {
+        line.withEnvironment("ANDROID_HOME", androidHome);
+      }
+      line.setExePath(FileUtil.toSystemDependentName(sdk.getHomePath() + "/bin/" + FlutterSdkUtil.flutterScriptName()));
+      if (workDir != null) {
+        line.setWorkDirectory(workDir.getPath());
+      }
+      if (!isDoctorCommand()) {
+        line.addParameter("--no-color");
+      }
+      line.addParameters(type.subCommand);
+      line.addParameters(args);
+    }
+    return line;
+  }
 
-    line.setExePath(FileUtil.toSystemDependentName(sdk.getHomePath() + "/bin/" + FlutterSdkUtil.flutterScriptName()));
+  /**
+   * Creates a FlutterWeb command line to run.
```

Comment:
remove this comment as it doesn't add any value beyond the method name.

Reply:
done